eth/consensus/clique/clique.py

```diff
@@ -85,45 +90,16 @@ def __init__(self, base_db: AtomicDatabaseAPI, epoch_length: int = EPOCH_LENGTH)
             self._epoch_length,
         )
 
-    @to_tuple
-    def amend_vm_configuration(self, config: VMConfiguration) -> Iterable[VMFork]:
-        """
-        Amend the given ``VMConfiguration`` to operate under the rules of Clique consensus.
-        """
-        for pair in config:
-            block_number, vm = pair
-            vm_class = vm.configure(
-                extra_data_max_bytes=65535,
-                validate_seal=staticmethod(self.validate_seal),
-                create_execution_context=staticmethod(self.create_execution_context),
-                configure_header=configure_header,
-                _assign_block_rewards=lambda _, __: None,
-            )
-
-            yield block_number, vm_class
-
-    @staticmethod
-    def create_execution_context(header: BlockHeaderAPI,
-                                 prev_hashes: Iterable[Hash32],
-                                 chain_context: ChainContext) -> ExecutionContext:
-
+    @classmethod
+    def get_fee_recipient(cls, header: BlockHeaderAPI) -> Address:
         # In Clique consensus, the tx fee goes to the signer
         try:
-            coinbase = get_block_signer(header)
+            return get_block_signer(header)
         except ValueError:
-            coinbase = header.coinbase
-
-        return ExecutionContext(
-            coinbase=coinbase,
-            timestamp=header.timestamp,
-            block_number=header.block_number,
-            difficulty=header.difficulty,
-            gas_limit=header.gas_limit,
-            prev_hashes=prev_hashes,
-            chain_id=chain_context.chain_id,
-        )
+            return header.coinbase
 
-    def get_snapshot(self, header: BlockHeaderAPI) -> Snapshot:
+    #TODO does this need to be a public API, or just something for test introspection?
```

Comment:
It is *currently* only used in tests but geth has an entire RPC API dedicated for Clique that would need to access this e.g.

```
curl -X POST --data '{"jsonrpc":"2.0","method":"clique_getSnapshot","params":[],"id":1}' http://localhost:8545
curl -X POST --data '{"jsonrpc":"2.0","method":"clique_getSigners","params":[],"id":1}' http://localhost:8545
```

In fact, these APIs where very helpful to add Clique support :)
That's why I think `get_snapshot` should continue to be a public API on `CliqueConsensus`.